### PR TITLE
Add gitignore for Modelica

### PR DESCRIPTION
**Pull request by dietmarw**
**Reasons for making this change:**

I'm a long term contributor to the [Modelica Language Specification](https://modelica.org) and Modelica could benefit greatly if there was a ready to use gitignore boiler plate file. All the development of Modelica projects happens on GitHub: https://github.com/modelica

**Links to documentation supporting these rule changes:**
- https://specification.modelica.org/master/

If this is a new template:

 - https://modelica.org
 - https://github.com/modelica

### CODE REVIEW

**Review by NaNOteh — APPROVED** [review]

File: Modelica.gitignore [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+dsres.txt
+dymosim*
+request
+stat
+status
+stop
+success
+*.
+
+## Files generated by OS X
```
Comment (wirecat) [inline Modelica.gitignore:44]: 👋 Hello, could you please remove these OS X specific lines?
That currently goes against our guidelines. There is an OS X specific gitignore in the globals folder that contains this.

Comment (dietmarw) [reply Modelica.gitignore]: Done :-)

**Comment by wirecat** [comment]
Thank you for the updates!

**Comment by dietmarw** [comment]
Thanks for the merge. :-)

**Pull request merged**